Newsletters: add 98 (2020-05-20)

harding opened the pull request:
- [x] Lede, action items, news, releases/RCs, any additional notable merges @harding 
- [x] Update the release/RC section Tuesday afternoon @harding 
- [x] Changes to services and client software @bitschmidty 
- [x] `Bitcoin Core #18877` @jnewbery 
- [x] `C-Lightning #3614` @dongcarl

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,167 @@
+  There's no need[^getheaders-details] to send more than one `getheaders`
+  request and Bitcoin Core doesn't send more than one block announcement
+  in an `inv` message, so this change should have no effect on Bitcoin
+  Core nodes who connect to Bitcoin Core peers; it's only expected
```
suggestions: 
- s/who/that/
- s/it's/its/ (or "the")

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,167 @@
+
+- [Eclair #1395][] updates the route pathfinding used by Eclair to
+  factor in channel balances and to use [Yen's algorithm][].  Testing
+  quoted in the PR description says, "The new algorithm consistently
```
"Testing" seems awkward as a subject.

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,167 @@
+    elegance] the "elegance" of a simplified version of the protocol
+    that uses three transactions.  Dmitry Petukhov [posted][petukhov
+    tla+] about a [specification][sas tla+ spec] he'd written for the
+    protocol in the [TLA<sup>+</sup> formal specification language][tla+
```
perhaps add "(Temporal Logic of Actions)" between "TLA+" and "formal" to describe the acronym without the reader needing to leave the newsletter.

harding replied:
In general I do think we should expand unfamiliar abbreviations.  In this case, I don't think it's useful---either people are familiar with the language or not.  Those who are familiar with it either know what the abbreviation stands for or don't care about it.  Those who are not familiar with TLA+ probably want to know more about it than the expansion of its abbreviation, so they're going to click the link anyway.

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,167 @@
+  There's no need[^getheaders-details] to send more than one `getheaders`
+  request and Bitcoin Core doesn't send more than one block announcement
+  in an `inv` message, so this change should have no effect when Bitcoin
+  Core nodes connect to Bitcoin Core peers; it's only expected
```
s/it's/its/ (as this is possessive and not a contraction)

Considering the sentence length, could also break into 2 sentences here with s/; it's/. The/

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,167 @@
+
+- [Eclair #1395][] updates the route pathfinding used by Eclair to
+  factor in channel balances and to use [Yen's algorithm][].
+  The PR description say that "the new algorithm consistently
```
s/say/says/ (or "According to the PR description: ")

jnewbery commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,177 @@
+- **Evaluate proposed changes to BIP341 taproot transaction digest:** as
+  described in [last week's newsletter][news97 spk commit], there has
+  been a request for [taproot][topic taproot] signatures to make an
+  additional commitment to all the scriptPubKeys of the UTXOs being
```
nit: s/all the scriptPubKeys of the UTXOs/the scriptPubKeys of all the UTXOs/ sounds slightly better to me (the first sounds to me a bit like a UTXO could include multiple scriptPubKeys).

jnewbery commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,177 @@
+    that uses three transactions.  Dmitry Petukhov [posted][petukhov
+    tla+] about a [specification][sas tla+ spec] he'd written for the
+    protocol in the [TLA<sup>+</sup> formal specification language][tla+
+    lang], helping to verify the correctness of the protocol (to a
```
This parenthetical is a little awkward. I think it can be removed.

harding commented:
Made suggested edits (thanks @jonatack and @jnewbery!) and made a few small edits to the merge summary written by @jnewbery (thanks!).

jnewbery commented:
ACK 99ea5a86ad11219c63d722f415c76d7aa5f0048c. All looks good so far. Thanks for the quick turnaround, @harding.

jnewbery commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -62,7 +62,25 @@ changes to popular Bitcoin infrastructure software.
+
+- **Lightning Loop using multipath payments:**
+  Lightning Labs' Lightning Loop product [now uses][lightning loop mpp blog]
+  [multipath payments][topic multipath payments] to convert onchain funds into
```
having two links immediately next to each other makes it difficult to see that they're separate links (especially when it's over a link break, as this appears in my browser). Is there a way to rearrange this sentence to avoid that?

bitschmidty replied:
Agreed! Fixed.

jnewbery commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -62,7 +62,25 @@ changes to popular Bitcoin infrastructure software.
+- **Blockstream Satellite 2.0 supports initial block download:**
+  Blockstream [outlines version 2.0 upgrades][blockstream satellite v2 blog] to
+  their satellite service which include expanded Asia-Pacific coverage and
+  additional bandwidth, enabling a full node to complete an initial sync
```
_complete_ an initial sync _completely_ fees a bit repetitive. Consider ".. enabling a full node to do an initial sync using just the satellite feed."

bitschmidty replied:
Tweaked. Thanks for the feedback.

adamjonas commented:
ACK f9a2f1e. Looks good!

harding commented:
Thanks @bitschmidty for your section and @dongcarl for your merge summary!  I made a few edits, most of which I think are self-explanatory, but two to Schmidty's section that might warrant additional explanation:

- I added a mention about the protocol upgrade in the Blockstream Satellite section for two reasons: (1) so that it is clear to existing satellite users that they need to upgrade to get the ability to IBD over satellite and (2) because, from what I've heard, the new protocol is pretty nifty, so I wanted to mention it explicitly.

- I moved the comma-delimited parenthetical in the Copay description to a separate sentence both because it was a bit awkward (and I *never* (ever) write awkward parentheticals) and because I wanted to make it clearer to readers that waiting four hours to fee bump is not a requirement of CPFP, it's a decision by the Copay authors.

As always, please feel free to revert or edit any changes.  Thanks!

bitschmidty commented:
> added a mention about the protocol upgrade in the Blockstream Satellite
SGTM

> moved the comma-delimited parenthetical in the Copay description to a separate sentence
I parenthetical too much. Change looks good.

jnewbery commented:
All changes look good to me. Thanks Harding!

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+- **Evaluate proposed changes to BIP341 taproot transaction digest:** as
+  described in [last week's newsletter][news97 spk commit], there has
+  been a request for [taproot][topic taproot] signatures to make an
+  additional commitment to the scriptPubKeys of the all the UTXOs being
```
s/of the all/of all/

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+    than existing protocols, it saves on transaction fees (both by using
+    less block space and potentially by requiring less urgency for its
+    settlement transactions), it only requires consensus-enforced
+    timelocks on one of the chains in a cross-chain swap, and that it
```
perhaps remove this "that"

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+    settlement transactions), it only requires consensus-enforced
+    timelocks on one of the chains in a cross-chain swap, and that it
+    doesn't depend on any new security assumptions or Bitcoin consensus
+    changes.  If taproot is adopted, it can be used even more privately
```
ambiguous "it"; perhaps "this proposal (or protocol) would allow it to be used even more privately..."

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+- **Blockstream Satellite 2.0 supports initial block download:**
+  Blockstream [outlines version 2.0 upgrades][blockstream satellite v2 blog] to
+  their satellite service which include expanded Asia-Pacific coverage,
+  additional bandwidth, and an updated protocol that enabling a full node to complete an initial sync
```
remove "that", or s/enabling/enables/

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+
+- **Breez wallet enables spontaneous payments:**
+  [Version 0.9][breez 0.9] of Breez wallet adds the ability to send spontaneous
+  payments to Lightning nodes that support keysend.
```
maybe?: [keysend payments][topic spontaneous payments].

bitschmidty replied:
Only wanted to avoid because we used it a couple bullets earlier.

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+  since the first release candidate.
+
+- [LND 0.10.1-beta.rc1][] is the first release candidate for the next
+  maintenace release of LND.
```
maintenance

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+
+- [Bitcoin Core #18877][] is the first step towards support for serving
+  [compact block filters][topic compact block filters] on the P2P
+  network, as specified in [BIP157][].  Nodes that enabled the compact block filter index
```
s/enabled/enable/

jonatack commented on _posts/en/newsletters/2020-05-20-newsletter.md:
```diff
@@ -0,0 +1,181 @@
+  `getdata` requests that specify an unknown type of data.  The new
+  logic will also ignore requests for types of data that aren't expected
+  to be sent over the current connection, such as requests for
+  transactions on block-relay-only connections.
```
Perhaps describe or mention the change from previous behavior.

harding commented:
@jonatack thanks for that extra review!

bitschmidty commented:
Squashed and merged!

👍 team! Thanks to @harding @jnewbery @dongcarl for authoring and @adamjonas @jonatack for reviewing.
